### DISCUSSION TITLE
Installing on Netbeans 14

### DISCUSSION THREAD

**Discussion by jshster** [Chris2011/minifierbeans#98] [General]
Just shifting from Issues to Discussion.
Can minifierbeans be installed on Netbeans 14?
I'm on a Windows 10 machine.  JDK is 16.  

Here is the plugin details I had installed on Netbeans 12.3: 
Version: 2.5.1 Source: 1560272403_org-netbeans-minify.nbm  
Plugin Description
See the [changelog](https://github.com/Chris2011/minifierbeans/blob/master/Changelog.md) for all changes and the [readme](https://github.com/Chris2011/minifierbeans/blob/master/Readme.md) for all information.

Anything else you need to know?

**Comment by Chris2011** [comment]
> Can minifierbeans be installed on Netbeans 14?
Yes, it can, I have it.

The newest version is 3.4.0, see the releases. Which plugin version did you tried for NetBeans 14? 12.3. is not relevant.

**Comment by jshster** [comment]
I downloaded the latest release yesterday but clearly it requires compiling or something to get it installed - or perhaps I'm not aware of the manual installation steps.  
I normally go to Tools | Plugins | Downloaded (tab) and then click Add Plugins.  This then allows you to either select an .nbm or .jar file from your file system but the only file it detects is gson-2.2.3.jar.  The other 3 files (code-minify.jar, html-minify.jar & image-compressor.jar) if you click on them Netbeans ignores them completely.  

Sorry if I'm being stupid here.  Feels like I'm missing a step or missing files.

**Reply by Chris2011** [reply]
You already said that you downloaded the nbm file, there is no need to have any jar files.

Ok do step by step:
- Download latest nbm release which is 3.4.0
- Open NetBeans and go to tools -> plugins -> Add plugin and select the nbm file, just the nbm file.
- Click on install and maybe create a screenshot of the error to show me what happens there. If it says smth with pack200, this is what I guess what happened there, just try to open NetBeans with a JDK < 14 for example 11. Just to install the plugin because pack200 was removed from JDK 14. But first things first.

**Reply by jshster** [reply]
My bad, I never realised that you click on Release and the download of the nbm file is available there.  Every other repo I've used I've gone to the Code | Download Zip option!!!  
So sorry to have bothered you.

**Reply by Chris2011** [reply]
I'm glad that it worked out :). In general, if you just download the zip file, it is always just the source. It is like git clone but without using it. After you download the zip, e.g. an NetBeans plugin, you can open it with NetBeans as a project and you need to build it.

**Reply by kaakemp** [reply]
Hi,
I'm trying to install the nbm file on Netbeans 14.
I get a validation warning saying that the plugin is unsigned and might be dangerous.

![netbeans-minify-msg](https://user-images.githubusercontent.com/38323735/188013904-d5cc0d52-e51f-4896-9147-5c2e69f77690.jpg)

**Reply by Chris2011** [reply]
Yes it is just a warning, that this plugin is unsiged. As you can see it exists for many years and you can have a look into the code. There is no malicious code. So it is fine to use it. As mentioned one day I will bring it to the plugin portal but not now.

**Comment by jshster** [comment]
Unless the plugin has gone through the official plugin portal you will see this message.  Just click continue and it will install.

**Comment by kaakemp** [comment]
Thanks, I’ve added it to my Apache NetBeans IDE 14I see an error in the Minification in progress tab:‘”node”’ is not recognized as an internal or external command, operable program or batch file. Although the popup reports success, there is no change on disk for files.Is there something I don’t have set up for the NetBeans IDE? 
![image](https://user-images.githubusercontent.com/38323735/188687560-dc2820b1-b2b5-48e1-8435-4445bd465df5.png)

**Reply by Chris2011** [reply]
I know that it can sounds weird, but first check whether you had node installed and also added to the PATH variable of Windows. You can check this by opening the CMD and write `node -v`. Next is, check the options in NetBeans -> Tools -> Options -> HTML5/JS -> Node whehter you already added node there or not. If everything seems fine, let me know but please create a new discussion. If the process doesn't give you the expected result, please create a new Issue.

In general, you should get the minified versionof your file: myfile.js -> minifierbeans -> myfile.min.js. If you've left the default options. If this is not the case, please create an issue with exact steps what you did (Single file, project type, environment information, versions, etc.)
